public/js/p3/widget/app/GeneTree.js

```diff
@@ -0,0 +1,615 @@
+          td.genomeRecord = lrec;
+          td.innerHTML = "<div class='libraryrow'>" + this.makeGenomeGroupName() + '</div>';
+          domConstruct.create('td', { innerHTML: '' }, tr);
+          var td2 = domConstruct.create('td', { innerHTML: "<i class='fa icon-x fa-1x' />" }, tr);
+          if (this.addedGenomes < this.startingRows) {
+            this.genomeTable.deleteRow(-1);
+          }
+          var handle = on(td2, 'click', lang.hitch(this, function (evt) {
+            // console.log("Delete Row");
+            domConstruct.destroy(tr);
+            this.decreaseGenome('genome_group', newGenomeIds);
+            if (this.addedGenomes < this.startingRows) {
+              var ntr = this.genomeTable.insertRow(-1);
+              domConstruct.create('td', { innerHTML: "<div class='emptyrow'></div>" }, ntr);
+              domConstruct.create('td', { innerHTML: "<div class='emptyrow'></div>" }, ntr);
+              domConstruct.create('td', { innerHTML: "<div class='emptyrow'></div>" }, ntr);
+            }
+            handle.remove();
+          }));
+          this.increaseGenome('genome_group', newGenomeIds);
+        }
+      }));
+
+      // console.log(lrec);
+    },
+
+    onSubmit: function (evt) {
+      var _self = this;
+
+      evt.preventDefault();
+      evt.stopPropagation();
+      if (this.validate()) {
+        var values = this.getValues();
+
+        // if (numUserGenome > 1 && values.reference_genome_index > 0) {
+        domClass.add(this.domNode, 'Working');
+        domClass.remove(this.domNode, 'Error');
+        domClass.remove(this.domNode, 'Submitted');
+
+        if (window.App.noJobSubmission) {
+          var dlg = new Dialog({
+            title: 'Job Submission Params: ',
+            content: '<pre>' + JSON.stringify(values, null, 4) + '</pre>'
+          });
+          dlg.startup();
+          dlg.show();
+          return;
+        }
+        this.submitButton.set('disabled', true);
+        window.App.api.service('AppService.start_app', [this.applicationName, values]).then(function (results) {
+          // console.log("Job Submission Results: ", results);
+          domClass.remove(_self.domNode, 'Working');
+          domClass.add(_self.domNode, 'Submitted');
+          _self.submitButton.set('disabled', false);
+          registry.byClass('p3.widget.WorkspaceFilenameValidationTextBox').forEach(function (obj) {
+            obj.reset();
+          });
+        }, function (err) {
+          // console.log("Error:", err)
+          domClass.remove(_self.domNode, 'Working');
+          domClass.add(_self.domNode, 'Error');
+          _self.errorMessage.innerHTML = err;
+        });
+        // } else {
+        domClass.add(this.domNode, 'Error');
+        // console.log("Form is incomplete");
+        // }
+
+      } else {
+        domClass.add(this.domNode, 'Error');
+        // console.log("Form is incomplete");
+      }
+    },
```

Comment:
Is there any reason to overwrite the AppBase onSubmit()? If possible, I would like to reuse the onSubmit() of the base class so that we can control the submission in a single place.

Reply:
I removed onSubmit().  No reason to overwrite AppBase onSubmit().